playback/src/player.rs

```diff
@@ -489,7 +493,10 @@ impl PlayerState {
         match *self {
             Stopped | EndOfTrack { .. } | Paused { .. } | Loading { .. } => false,
             Playing { .. } => true,
-            Invalid => panic!("invalid state"),
+            Invalid => {
+                error!("PlayerState is_playing: invalid state");
```

Comment:
We should consider moving these to the `debug` level. As-is they are very technical.

Reply:
I would prefer to keep anything that replaces a `panic!` with a message and an `exit(1)` as an `error!` the reason being that otherwise basically if we use `debug!` we're just silently failing. I'm all for graceful exits but going from blowing up and  puking (panic) to just exiting silently (most people don't run in debug all the time) is a little too far IMHO.

Reply:
That's true. Then perhaps we could make the error message a bit more user-friendly.

Reply:
> That's true. Then perhaps we could make the error message a bit more user-friendly.

I'm really not sure how I could do that? `player` is pretty spaghettified and not the easiest for my pea brain to follow. I'm not even certain all of those `invalid` states are even reachable. I have to really dig super deep in everything that `player` touches which is basically everything.

Reply:
> We can go either way but I don't like bothering regular users with cryptic messages.

I'm not really sure these messages are a bother at all? Either nothing is going wrong and you get no messages from this code because it only squawk when there is an actual error, or librespot runs into an error and spits out a short and concise error message that's easy for someone, either a dev or a user, to search up in the codebase for so they can figure out what the problem is. These error points have a potential to not reveal much info, throwing away *more* info I don't think is the way to go. Remember these are fatal errors. It's not like librespot will continue to run and spit out potentially more useful info after these messages. librespot will output the message and promptly commit suicide.

Reply:
It's not my intention to turn this into a bike shed. Although my view remains that it'd be easy to make them a bit friendlier, these messages were already there before your PR so I'm fine with doing this separately in the future, if anyone else would care to spend the time on it. Guess our time is better spent on other things!

Reply:
So we can agree to disagree on this and just deal with it later? Can I Resolve this (for now at least)?

Reply:
Yes.